enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,434 @@
+4. The project admin creates an HTTPRoute, specifying the desired host name, the Gateway from Step 3, and the target service in the project admin's namespace.
+5. The service end-user connects to the host name specified in the HTTPRoute from Step 4 and receives a response from the service that the HTTPRoute targets.
+
+### API Extensions
+
+This enhancement imports the v1beta1 versions of the following CRDs into OpenShift:
+
+* [GatewayClass](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/).
+* [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/).
+* [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/).
+
+Before this enhancement, DNSRecord CRs were not reconciled unless they were in
+the "openshift-ingress-operator" namespace and had an associated
+IngressController.  This enhancement enables reconciliation of DNSRecord CRs in
+the "openshift-ingress" namespace.
+
+This enhancement adds a controller that creates DNSRecord CRs for specially
+annotated services in the "openshift-ingress" namespace.  This controller is
+used to configure DNS for the LoadBalancer-type service associated with a
+Gateway.
+
+Tentatively, this enhancement may define an additional CRD for specifying
+configuration such as the following:
+
+* Whether Gateway API/ingress is enabled.
+* Whether Istio API/mesh is enabled.
+* Whether to create a default Gateway.
+* Istiod options, such as profile and addons (Jaeger, Kiali, Grafana, Prometheus).
+* OSSM automatic deployments.
+
+See [Open Questions](#open-questions) for some related discussion.
+
+### Implementation Details/Notes/Constraints
+
+We have several constraints related  design and release-timing constraints to consider.
+
+#### Ownership of the ServiceMeshControlPlane
+
+OpenShift Service Mesh is configured through the ServiceMeshControlPlane CR,
+both for ingress use-cases and for mesh use-cases.  [The existing installation
+procedure for Service
+Mesh](https://docs.openshift.com/container-platform/4.11/service_mesh/v2x/ossm-create-smcp.html#ossm-control-plane-deploy-cli_ossm-create-smcp)
+instructs the reader to create a ServiceMeshControlPlane named "basic" in the
+"istio-system" namespace.  To ensure backwards compatibility, the Ingress
+Operator must do one of the following:
+
+* Use the same name and namespace: "istio-system/basic".
+* Migrate an existing SMCP from "istio-system" to "openshift-ingress".
+* Provide an API to tell the Ingress Operator what SMCP to use.
+
+Placement, possible migration, and ownership of the ServiceMeshControlPlane need
+to be clarified in this enhancement.  This is an open design question.
+
+#### Managing the ServiceMeshMemberRoll
+
+OSSM 2.3 only watches resources in namespaces specified in an
+ServiceMeshMemberRoll CR associated with the ServiceMeshControlPlane.  We hope
+that this limitation will be removed from OSSM in time for this enhancement;
```

Comment:
I think https://issues.redhat.com/browse/OSSM-924 is where Daniel is doing his prototyping work for this.